pkg/cmd/roachtest/option/options.go

```diff
@@ -174,6 +174,17 @@ func SkipInit(opts interface{}) {
 	}
 }
 
+// Tag sets a process tag when stopping processes. Useful if we want
+// to kill a cockroach process that was started with `settings.TagOption`.
```

Comment:
Is `ProcessTag` actually used by the kill script? I couldn't find any reference sites other than `getClusterFromCache`.

Reply:
It's fairly subtle, but this is how it works:

* `c.Stop` is called with the `Tag` option, causing `RoachprodOpts.ProcessTag` to be set (the code you're commenting on)
* Those `RoachprodOpts` are passed to `roachprod.Stop`:

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/cmd/roachtest/cluster.go#L2397

* The process tag is passed when loading the cluster from the cache:

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/roachprod.go#L912

* This means that the `*install.SyncedCluster` instance will create a `ClusterSettings` object with that tag, and `SyncedCluster` embeds `ClusterSettings`:

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/install/cluster_settings.go#L112-L126

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/install/cluster_synced.go#L74-L80

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/install/cluster_synced.go#L62

* When the `Stop` call eventually calls `c.kill()`, the kill script will filter the process to kill based on the `roachprodEnvRegex`:

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/install/cluster_synced.go#L532-L535

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/install/cluster_synced.go#L543

* `roachprodEnvRegex` returns the value that should be set in the `ROACHPROD` environment variable on the process. It includes the label we set:

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/install/cluster_synced.go#L292-L294

https://github.com/cockroachdb/cockroach/blob/83a68a474a31b6ffc2d7c45d0097b859568e60dc/pkg/roachprod/install/cluster_synced.go#L270-L280

*****
TLDR: only a process started with the same tag should be killed when `c.Stop()` is called with `option.Tag()`.